lib/collection/src/shards/replica_set.rs

```diff
@@ -921,14 +922,17 @@ impl ShardReplicaSet {
             let error_count = errors.len();
 
             Err(CollectionError::service_error(format!(
-                "{} of {} shards failed with: {}",
-                error_count,
+                "{error_count} of {} shards failed with: {}",
                 success_count + error_count,
-                errors
-                    .into_iter()
-                    .map(|err| format!("{}", err))
-                    .collect::<Vec<_>>()
-                    .join(", ")
+                errors.into_iter().fold(String::new(), |mut msg, err| {
```

Comment:
I liked the `join` better! :cry: 

```rust
errors
    .into_iter()
    .map(|err| format!("{}", err))
    .collect::<Vec<_>>()
    .join(", ")
```

Reply:
I kinda do too... but it's a lot of extra allocations! 😢